pkg/kube/multicluster/secretcontroller.go

```diff
@@ -150,10 +150,12 @@ func NewController(kubeclientset kube.Client, namespace string, clusterID cluste
 		informer:            secretsInformer,
 	}
 
-	nsInformer := kubeclientset.KubeInformer().Core().V1().Namespaces().Informer()
-	_ = nsInformer.SetTransform(kube.StripUnusedFields)
-	nsLister := kubeclientset.KubeInformer().Core().V1().Namespaces().Lister()
-	controller.DiscoveryNamespacesFilter = filter.NewDiscoveryNamespacesFilter(nsLister, meshWatcher.Mesh().GetDiscoverySelectors())
+	if features.EnableEnhancedResourceScoping {
```

Comment:
I think we cannot do this? Because before we use discovery selector to limit the crd, we are already using it to limit the services.

This will bring a regression if one have already use discovery selector for services

Reply:
Isn't this influencing original service filter?

Reply:
IIUC This is also passed to kube controller

Reply:
Maybe folks depending on this feature (@harveyxia @hzxuzhonghu @kfaseela ) could add integration tests for these features. Then we won't have to guess if it works or not :-)